pkg/controller/instance/controller_reconcile.go

```diff
@@ -123,193 +182,102 @@ func (igr *instanceGraphReconciler) reconcileInstance(ctx context.Context) error
 		igr.state.ResourceStates[resourceID] = &ResourceState{State: ResourceStatePending}
 	}
 
-	// Reconcile resources in topological order
-	for _, resourceID := range igr.runtime.TopologicalOrder() {
-		if err := igr.reconcileResource(ctx, resourceID); err != nil {
-			return err
-		}
-
-		// Synchronize runtime state after each resource
-		if _, err := igr.runtime.Synchronize(); err != nil {
-			return fmt.Errorf("failed to synchronize reconciling resource %s: %w", resourceID, err)
-		}
+	// TODO (barney-s):
```

Comment:
I think this todo can be removed after https://github.com/kro-run/kro/pull/561/files#r2198966036

Reply:
ah yes. removed.